sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
Ah, I see what you mean, now.

However, I'm pretty sure that the problem addressed by `commonJSName` is completely different than the problem in #2797. As such, I don't think we're removing a solution to #2797. Let me elaborate.

### The problem addressed by `commonJSName`

I design my codebase (and potentially facades) exclusively for a "browser" environment. Therefore, I depend only on libraries that are meant to be used directly in a browser environment, without any processing. However, I'm unlucky because one such library is "too smart", and it behaves differently if it detects a Node.js environment.

`commonJSName` solves this problem by "restoring" the browser behavior of the library when it is run in a Node.js environment.

Can multiple loading specs solve this problem too? Possibly, but I don't think so. Even if I prepare my library for `@JSGlobal`+`@JSImport`, only one of those is selected to perform `fastOptJS`, as determined by `scalaJSModuleKind`. The same `-fastopt.js` is run both in the browser and in Node.js, so it cannot adapt to the "too smart" behavior of the library anyway.

What else can solve this problem? IMO any bundling tool for JS libraries, or, put another way, JS dependency manager. Depending on where libraries are fetched from, and how they are processed, it might be necessary or not to process them so that they all conform to one model (either all conform to CommonJS, or all conform to browser). `jsDependencies` is one such JS dependency manager, and `commonJSName` causes it to process some JS libraries so that they all follow the browser environment. It therefore makes sense that the `commonJSName` "processor" stays with the dependency manager that uses it, i.e., `jsDependencies`.

### The problem in #2797

Here the problem is that I want to write a library in a way that is agnostic to whether the final application uses the browser environment (`NoModule`) or a CommonJS environment (possibly with an associated bundler).

Can `commonJSName` solve this problem? Absolutely not, at least not in its current incarnation: `commonJSName` is tight to `jsDependencies`, and we know `jsDependencies` to be completely incompatible with the `CommonJSModule` module kind. So there's no way they can provide a solution to a problem that only manifests itself when we double-target `NoModule` and `CommonJSModule`.

### Conclusion

The problem described in #2797 and the problem addressed by `commonJSName` are completely different issues. And the solutions to either cannot solve the other problem. Hence, I don't think that removing `commonJSName` from core is a step backwards wrt. #2797. It is completely orthogonal.